woocommerce/woocommerce-ios
# Coupons: Simplify title for Edit Coupon view and enable updating discount type

Closes: #6912 


### Description

Currently, we're specifying discount type on the title of the Edit Coupon view, which is lengthy that can be cut off on small devices. Also, this indicates that the coupon's discount type cannot be updated, which doesn't align with the feature on WP Admin.

This PR addresses both of the issues:
- Make the title of the Edit Coupon view generic by using "Edit Coupon" or "Create Coupon" - depending on the flow the screen is displaying.
- Enable updating discount type:
  - Since we're using the custom bottom sheet design to present the list of discount types with descriptions, a new `BottomSheetListSelectorCommand` type is added for discount type.
  - Update `ImageAndTitleAndTextTableViewCell` with a checkmark for the accessory view if the item it represents is selected.
  - Present the discount type bottom sheet from the Edit coupon view. Since we want to display the bottom sheet that covers just half the screen on iPhone while showing a popover pointing from the correct view, my solution was to present the sheet the UIKit way on iPhone and embed it in a popover on iPad. This should be updated with a [`UISheetPresentationController`](https://developer.apple.com/documentation/uikit/uisheetpresentationcontroller) wrapper when we drop support for iOS 14.

### Testing instructions

- Make sure to enable coupon management in experimental features.
- Navigate to Menu > Coupons > select a coupon with a supported discount type.
- Select the ellipsis button > Edit Coupon.
- Notice that the presented screen's title is "Edit Coupon", and a new field for updating discount type is on the top.
- Tap Discount type, and notice that a bottom sheet is presented if you're on an iPhone. On iPads, the content is displayed in a popover presented from the Discount type row.
- When switching between discount types, the Amount field should be updated accordingly. Saving the updates should succeed.

### Screenshots

On iPhone:

https://user-images.githubusercontent.com/5533851/170169510-e100b806-41f4-438a-9d64-de4d9eb79571.mp4

On iPad:

https://user-images.githubusercontent.com/5533851/170171297-cf46f645-78b9-4baf-b97c-61eca05d4786.mov



---
